exec/java-exec/src/main/java/org/apache/drill/exec/ops/OperatorContextImpl.java

```diff
@@ -130,6 +148,27 @@ public OperatorStats getStats() {
     return stats;
   }
 
+  public <RESULT> ListenableFuture<RESULT> runCallableAs(final UserGroupInformation proxyUgi,
+                                                         final Callable<RESULT> callable) {
+    synchronized (this) {
```

Comment:
But the worker pool is unbounded, and this method could potentially be misused by submitting a large number of tasks. Is that fine?